### PR TITLE
docs: clarify native addressing

### PR DESCRIPTION
**Pull request by lidel**

### CODE REVIEW

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/how-to/address-ipfs-on-web.md`:
- The following link returned HTTP status code 302 (Found):
  https://tools.ietf.org/html/rfc1034
- The following link returned HTTP status code 301 (Moved Permanently):
  https://dweb.link/ipfs/QmT5NvUtoM5nWFfrQdVrFtvGfKFmG7AHE8P34isapyhCxX/wiki/Mars.html
- The following link returned HTTP status code 302 (Found):
  https://tools.ietf.org/html/rfc4648
- The following link returned HTTP status code 301 (Moved Permanently):
  https://github.com/ipfs-shipyard/ipfs-companion
- ‘in brief’ might be wordy. Consider a shorter alternative.
  ```
  ...roken. Don't do that. # Dweb addressing in brief In IPFS, addresses (for con...
                                             ^^^^^^^^
  ```
- ‘by the use of’ might be wordy. Consider a shorter alternative.
  ```
  ... term, these issues should be mitigated by the use of opportunistic protocol...
                                             ^^^^^^^^^^^^^
  ```
- Possible missing comma found.
  ```
  ...llback when no native implementation is available in order to ensure a smoot...
                                             ^^^^^^^^^
  ```
- Consider a shorter alternative to avoid wordiness.
  ```
  ...n no native implementation is available in order to ensure a smooth, backwar...
                                             ^^^^^^^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...se one-liners for most common use cases Known issues Some browsers and other...
                                             ^^^^^
  ```
- Possible typo detected.
  ```
  ...r Base36 can be used for longer ED25519 libp2p keys. See the next section to...
                                             ^^^^^^
  ```
- Possible missing article found.
  ```
  ...ves the need for manual CID conversion. Request for a content path sent to t...
                                             ^^^^^^^
  ```
- This word has been used in one of the immediately preceding sentences. Using a synonym could make your text more interesting to read, unless the repetition is intentional.
  ```
  ...of any necessary encoding conversion if needed: To illustrate, opening the C...
                                             ^^^^^^
  ```
- Possible missing comma found.
  ```
  ...Manual — use cid.ipfs.io or the command line One can also do the conversion ...
                                             ^^^^
  ```
- Do not use a colon (:) before a series that is introduced by a preposition (‘with’). Remove the colon or add a noun or a noun phrase after the preposition.
  ```
  ...teway  (opens new window) HTTP URL, but with: protocol scheme replaced by ip...
                                             ^^^^^
  ```
- Did you mean “Discussions about”?
  ```
  ...ackground on address scheme discussions Discussions around IPFS addressing h...
                                             ^^^^^^^^^^^^^^^^^^
  ```
- The adjective ‘good’ can be overused, consider replacing it with a synonym.
  ```
  ...any lengthy GitHub issue threads, but a good summary can be found in this PR...
                                             ^^^^
  ```
- This phrase is duplicated. You should probably use “IPFS Companion” only once.
  ```
  ...found in this PR  (opens new window). # IPFS Companion IPFS Companion  (open...
                                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  ```
- A comma may be missing after the conjunctive/linking adverb ‘Currently’.
  ```
  ... and other content-addressed protocols. Currently investigated approaches in...
                                             ^^^^^^^^^
  ```

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/how-to/address-ipfs-on-web.md`:
- The following link returned HTTP status code 302 (Found):
  https://tools.ietf.org/html/rfc1034
- The following link returned HTTP status code 301 (Moved Permanently):
  https://dweb.link/ipfs/QmT5NvUtoM5nWFfrQdVrFtvGfKFmG7AHE8P34isapyhCxX/wiki/Mars.html
- The following link returned HTTP status code 302 (Found):
  https://tools.ietf.org/html/rfc4648
- The following link returned HTTP status code 301 (Moved Permanently):
  https://github.com/ipfs-shipyard/ipfs-companion
- ‘in brief’ might be wordy. Consider a shorter alternative.
  ```
  ...roken. Don't do that. # Dweb addressing in brief In IPFS, addresses (for con...
                                             ^^^^^^^^
  ```
- ‘by the use of’ might be wordy. Consider a shorter alternative.
  ```
  ... term, these issues should be mitigated by the use of opportunistic protocol...
                                             ^^^^^^^^^^^^^
  ```
- Possible missing comma found.
  ```
  ...llback when no native implementation is available in order to ensure a smoot...
                                             ^^^^^^^^^
  ```
- Consider a shorter alternative to avoid wordiness.
  ```
  ...n no native implementation is available in order to ensure a smooth, backwar...
                                             ^^^^^^^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...se one-liners for most common use cases Known issues Some browsers and other...
                                             ^^^^^
  ```
- Possible typo detected.
  ```
  ...r Base36 can be used for longer ED25519 libp2p keys. See the next section to...
                                             ^^^^^^
  ```
- Possible missing article found.
  ```
  ...ves the need for manual CID conversion. Request for a content path sent to t...
                                             ^^^^^^^
  ```
- This word has been used in one of the immediately preceding sentences. Using a synonym could make your text more interesting to read, unless the repetition is intentional.
  ```
  ...of any necessary encoding conversion if needed: To illustrate, opening the C...
                                             ^^^^^^
  ```
- Possible missing comma found.
  ```
  ...Manual — use cid.ipfs.io or the command line One can also do the conversion ...
                                             ^^^^
  ```
- Do not use a colon (:) before a series that is introduced by a preposition (‘with’). Remove the colon or add a noun or a noun phrase after the preposition.
  ```
  ...teway  (opens new window) HTTP URL, but with: protocol scheme replaced by ip...
                                             ^^^^^
  ```
- Did you mean “Discussions about”?
  ```
  ...ackground on address scheme discussions Discussions around IPFS addressing h...
                                             ^^^^^^^^^^^^^^^^^^
  ```
- The adjective ‘good’ can be overused, consider replacing it with a synonym.
  ```
  ...any lengthy GitHub issue threads, but a good summary can be found in this PR...
                                             ^^^^
  ```
- This phrase is duplicated. You should probably use “IPFS Companion” only once.
  ```
  ...found in this PR  (opens new window). # IPFS Companion IPFS Companion  (open...
                                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  ```
- A comma may be missing after the conjunctive/linking adverb ‘Currently’.
  ```
  ... and other content-addressed protocols. Currently investigated approaches in...
                                             ^^^^^^^^^
  ```

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/how-to/address-ipfs-on-web.md`:
- The following link returned HTTP status code 301 (Moved Permanently):
  https://dweb.link/ipfs/QmT5NvUtoM5nWFfrQdVrFtvGfKFmG7AHE8P34isapyhCxX/wiki/Mars.html
- ‘in brief’ might be wordy. Consider a shorter alternative.
  ```
  ...roken. Don't do that. # Dweb addressing in brief In IPFS, addresses (for con...
                                             ^^^^^^^^
  ```
- ‘by the use of’ might be wordy. Consider a shorter alternative.
  ```
  ... term, these issues should be mitigated by the use of opportunistic protocol...
                                             ^^^^^^^^^^^^^
  ```
- Possible missing comma found.
  ```
  ...llback when no native implementation is available in order to ensure a smoot...
                                             ^^^^^^^^^
  ```
- Consider a shorter alternative to avoid wordiness.
  ```
  ...n no native implementation is available in order to ensure a smooth, backwar...
                                             ^^^^^^^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...se one-liners for most common use cases Known issues Some browsers and other...
                                             ^^^^^
  ```
- Possible typo detected.
  ```
  ...r Base36 can be used for longer ED25519 libp2p keys. See the next section to...
                                             ^^^^^^
  ```
- Possible missing article found.
  ```
  ...ves the need for manual CID conversion. Request for a content path sent to t...
                                             ^^^^^^^
  ```
- This word has been used in one of the immediately preceding sentences. Using a synonym could make your text more interesting to read, unless the repetition is intentional.
  ```
  ...of any necessary encoding conversion if needed: To illustrate, opening the C...
                                             ^^^^^^
  ```
- Possible missing comma found.
  ```
  ...Manual — use cid.ipfs.io or the command line One can also do the conversion ...
                                             ^^^^
  ```
- Do not use a colon (:) before a series that is introduced by a preposition (‘with’). Remove the colon or add a noun or a noun phrase after the preposition.
  ```
  ...teway  (opens new window) HTTP URL, but with: protocol scheme replaced by ip...
                                             ^^^^^
  ```
- Did you mean “Discussions about”?
  ```
  ...ackground on address scheme discussions Discussions around IPFS addressing h...
                                             ^^^^^^^^^^^^^^^^^^
  ```
- The adjective ‘good’ can be overused, consider replacing it with a synonym.
  ```
  ...any lengthy GitHub issue threads, but a good summary can be found in this PR...
                                             ^^^^
  ```
- This phrase is duplicated. You should probably use “IPFS Companion” only once.
  ```
  ...found in this PR  (opens new window). # IPFS Companion IPFS Companion  (open...
                                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  ```
- A comma may be missing after the conjunctive/linking adverb ‘Currently’.
  ```
  ... and other content-addressed protocols. Currently investigated approaches in...
                                             ^^^^^^^^^
  ```

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/how-to/address-ipfs-on-web.md`:
- The following link returned HTTP status code 301 (Moved Permanently):
  https://dweb.link/ipfs/QmT5NvUtoM5nWFfrQdVrFtvGfKFmG7AHE8P34isapyhCxX/wiki/Mars.html
- ‘in brief’ might be wordy. Consider a shorter alternative.
  ```
  ...roken. Don't do that. # Dweb addressing in brief In IPFS, addresses (for con...
                                             ^^^^^^^^
  ```
- ‘by the use of’ might be wordy. Consider a shorter alternative.
  ```
  ... term, these issues should be mitigated by the use of opportunistic protocol...
                                             ^^^^^^^^^^^^^
  ```
- Possible missing comma found.
  ```
  ...llback when no native implementation is available in order to ensure a smoot...
                                             ^^^^^^^^^
  ```
- Consider a shorter alternative to avoid wordiness.
  ```
  ...n no native implementation is available in order to ensure a smooth, backwar...
                                             ^^^^^^^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...se one-liners for most common use cases Known issues Some browsers and other...
                                             ^^^^^
  ```
- Possible typo detected.
  ```
  ...r Base36 can be used for longer ED25519 libp2p keys. See the next section to...
                                             ^^^^^^
  ```
- Possible missing article found.
  ```
  ...ves the need for manual CID conversion. Request for a content path sent to t...
                                             ^^^^^^^
  ```
- This word has been used in one of the immediately preceding sentences. Using a synonym could make your text more interesting to read, unless the repetition is intentional.
  ```
  ...of any necessary encoding conversion if needed: To illustrate, opening the C...
                                             ^^^^^^
  ```
- Possible missing comma found.
  ```
  ...Manual — use cid.ipfs.io or the command line One can also do the conversion ...
                                             ^^^^
  ```
- Do not use a colon (:) before a series that is introduced by a preposition (‘with’). Remove the colon or add a noun or a noun phrase after the preposition.
  ```
  ...teway  (opens new window) HTTP URL, but with: protocol scheme replaced by ip...
                                             ^^^^^
  ```
- Did you mean “Discussions about”?
  ```
  ...ackground on address scheme discussions Discussions around IPFS addressing h...
                                             ^^^^^^^^^^^^^^^^^^
  ```
- The adjective ‘good’ can be overused, consider replacing it with a synonym.
  ```
  ...any lengthy GitHub issue threads, but a good summary can be found in this PR...
                                             ^^^^
  ```
- This phrase is duplicated. You should probably use “IPFS Companion” only once.
  ```
  ...found in this PR  (opens new window). # IPFS Companion IPFS Companion  (open...
                                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  ```
- A comma may be missing after the conjunctive/linking adverb ‘Currently’.
  ```
  ... and other content-addressed protocols. Currently investigated approaches in...
                                             ^^^^^^^^^
  ```

File: docs/how-to/address-ipfs-on-web.md [hunk 1]
```diff
@@ -138,7 +138,7 @@ The multihash in CIDv1 is the same as in the original CIDv0.
 #### Manual — use cid.ipfs.io or the command line
 
 One can also do the conversion manually.
```
Comment (johnnymatthews) [inline docs/how-to/address-ipfs-on-web.md:140]:
```suggestion
One can also do the conversion manually.

```

File: docs/how-to/address-ipfs-on-web.md [hunk 2]
```diff
@@ -170,16 +170,33 @@ For a complete DNSLink guide, including tutorials, usage examples, and FAQs, che
...
+ipns://{cidv1-libp2p-key}/path/to/resource
+ipns://{dnslink-name}/path/to/resource?query=foo#fragment
 ```
 
-An IPFS URL does not retain the original path, but instead requires a conversion step to/from URI representation:
+::: tip
+
+Our North Star here is to reuse existing standards to maximize interop with existing user agents like browsers and CLI tools, so if something is not clear, HTTP URL rules apply.
+
+:::
```
Comment (johnnymatthews) [inline docs/how-to/address-ipfs-on-web.md:198]:
```suggestion
::: tip
Our main goal here is to reuse existing standards that maximize interoperability with existing user-agents like browsers and CLI tools. If something is not clear, HTTP URL rules apply.
:::
```

File: docs/how-to/address-ipfs-on-web.md [hunk 3]
```diff
@@ -189,12 +206,21 @@ Example:
 ipfs://bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfyavhwq/wiki/Vincent_van_Gogh.html
 ```
 
-::: tip
+::: warning Avoid case-sensitive CID in ipfs://
```
Comment (johnnymatthews) [inline docs/how-to/address-ipfs-on-web.md:210]:
```suggestion
```

File: docs/how-to/address-ipfs-on-web.md [hunk 4]
```diff
@@ -189,12 +206,21 @@ Example:
...
 
-Native URIs require a CID to be case-insensitive. Use of CIDv1 in Base32 is advised.
+Some user agents will force-lowercase the CID component of URL-like address.
+To ensure interop with existing libraries and software, use case-insensitive CID encoding. Use of CIDv1 in Base32 or Base36 is advised.
```
Comment (johnnymatthews) [inline docs/how-to/address-ipfs-on-web.md:213]:
```suggestion
```

File: docs/how-to/address-ipfs-on-web.md [hunk 5]
```diff
@@ -189,12 +206,21 @@ Example:
...
 :::
 
+### Turning native address to a canonical content path
+
+Every "URL" address can be turned back into a content path with ease:
+
+> `ipfs://{immutable-root}/path/to/resourceA` → `/ipfs/{immutable-root}/path/to/resourceA`  
+> `ipns://{mutable-root}/path/to/resourceB` → `/ipns/{mutable-root}/path/to/resourceB`
+
+
```
Comment (johnnymatthews) [inline docs/how-to/address-ipfs-on-web.md:223]:
```suggestion
```

**Review by johnnymatthews — APPROVED** [review]

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/how-to/address-ipfs-on-web.md`:
- The following link returned HTTP status code 301 (Moved Permanently):
  https://dweb.link/ipfs/QmT5NvUtoM5nWFfrQdVrFtvGfKFmG7AHE8P34isapyhCxX/wiki/Mars.html
- ‘in brief’ might be wordy. Consider a shorter alternative.
  ```
  ...roken. Don't do that. # Dweb addressing in brief In IPFS, addresses (for con...
                                             ^^^^^^^^
  ```
- ‘by the use of’ might be wordy. Consider a shorter alternative.
  ```
  ... term, these issues should be mitigated by the use of opportunistic protocol...
                                             ^^^^^^^^^^^^^
  ```
- Possible missing comma found.
  ```
  ...llback when no native implementation is available in order to ensure a smoot...
                                             ^^^^^^^^^
  ```
- Consider a shorter alternative to avoid wordiness.
  ```
  ...n no native implementation is available in order to ensure a smooth, backwar...
                                             ^^^^^^^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...se one-liners for most common use cases Known issues Some browsers and other...
                                             ^^^^^
  ```
- Possible typo detected.
  ```
  ...r Base36 can be used for longer ED25519 libp2p keys. See the next section to...
                                             ^^^^^^
  ```
- Possible missing article found.
  ```
  ...ves the need for manual CID conversion. Request for a content path sent to t...
                                             ^^^^^^^
  ```
- This word has been used in one of the immediately preceding sentences. Using a synonym could make your text more interesting to read, unless the repetition is intentional.
  ```
  ...of any necessary encoding conversion if needed: To illustrate, opening the C...
                                             ^^^^^^
  ```
- Possible missing comma found.
  ```
  ...Manual — use cid.ipfs.io or the command line One can also do the conversion ...
                                             ^^^^
  ```
- Did you mean “Discussions about”?
  ```
  ...ackground on address scheme discussions Discussions around IPFS addressing h...
                                             ^^^^^^^^^^^^^^^^^^
  ```
- The adjective ‘good’ can be overused, consider replacing it with a synonym.
  ```
  ...any lengthy GitHub issue threads, but a good summary can be found in this PR...
                                             ^^^^
  ```
- This phrase is duplicated. You should probably use “IPFS Companion” only once.
  ```
  ...found in this PR  (opens new window). # IPFS Companion IPFS Companion  (open...
                                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  ```
- A comma may be missing after the conjunctive/linking adverb ‘Currently’.
  ```
  ... and other content-addressed protocols. Currently investigated approaches in...
                                             ^^^^^^^^^
  ```

**Pull request merged**